io_buffer.c

```diff
@@ -252,6 +256,21 @@ rb_io_buffer_type_allocate(VALUE self)
     return instance;
 }
 
+VALUE
+rb_io_buffer_type_for(VALUE klass, VALUE string)
+{
+    VALUE instance = rb_io_buffer_type_allocate(klass);
+
+    struct rb_io_buffer *data = NULL;
+    TypedData_Get_Struct(instance, struct rb_io_buffer, &rb_io_buffer_type, data);
+
+    rb_str_locktmp(string);
```

Comment:
You need to check it's actually a String before calling this.
And also you need to pin it to avoid it moving when GC.compact/autocompact is used.

Reply:
Thanks I will fix it.

Reply:
https://github.com/ruby/ruby/pull/5115 should resolve this.